spec/integration/puppet_x/puppetlabs/scheduled_task/taskscheduler2_spec.rb

```diff
@@ -0,0 +1,177 @@
+      subject_count = subject.enum_task_names(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::ROOT_FOLDER, { :include_compatibility => [PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V1]}).count
+      ps_cmd = '(Get-ScheduledTask | ? { [Int]$_.Settings.Compatibility -eq 1 } | Measure-Object).count'
+      expect(subject_count).to be_same_as_powershell_command(ps_cmd)
+    end
+  end
+
+  describe '#delete' do
+    before(:all) do
+      @task_name = PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::ROOT_FOLDER + 'puppet_task_' + SecureRandom.uuid.to_s
+    end
+
+    after(:all) do
+      begin
+        PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2.delete(@task_name)
+      rescue => _details
+        # Ignore any errors
+      end
+    end
+
+    it 'should delete a task that exists' do
+      create_test_task(@task_name)
+
+      # Can't use URI as it is empty string on some OS.  Just construct the URI
+      # using path and name
+      ps_cmd = '(Get-ScheduledTask | ? { $_.TaskPath + $_.TaskName -eq \'' + @task_name + '\' } | Measure-Object).count'
+      expect(1).to be_same_as_powershell_command(ps_cmd)
+
+      subject.delete(@task_name)
+      expect(0).to be_same_as_powershell_command(ps_cmd)
+    end
+  end
+
+  describe 'create a task' do
+    before(:all) do
+      @task_name = create_test_task
+    end
+
+    after(:all) do
+      PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2.delete(@task_name)
+    end
+
+    let(:task_object) { subject.task(@task_name) }
+    let(:task_definition) { subject.definition(task_object) }
+
+    context 'given a test task fixture' do
+      it 'should be disabled' do
+        expect(task_definition.Settings.Enabled).to eq(false)
+      end
+
+      it 'should be V2 compatible' do
+        expect(subject.compatibility(task_definition)).to eq(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_COMPATIBILITY_V2)
+      end
+
+      it 'should have a single trigger' do
+        expect(subject.trigger_count(task_definition)).to eq(1)
+      end
+
+      it 'should have a trigger of type TimeTrigger' do
+        expect(subject.trigger(task_definition, 1).Type).to eq(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_TRIGGER_TIME)
+      end
+
+      it 'should have a single action' do
+        expect(subject.action_count(task_definition)).to eq(1)
+      end
+
+      it 'should have an action of type Execution' do
+        expect(subject.action(task_definition, 1).Type).to eq(PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_ACTION_EXEC)
+      end
+
+      it 'should have the specified action path' do
+        expect(subject.action(task_definition, 1).Path).to eq('cmd.exe')
+      end
+
+      it 'should have the specified action arguments' do
+        expect(subject.action(task_definition, 1).Arguments).to eq('/c exit 0')
+      end
+    end
+  end
+
+  describe 'modify a task' do
+    before(:all) do
```

Comment:
Same here for `:each` instead of `:all` ... if more tests are added, you end up sharing the task which is probably undesirable.

Reply:
Done